File: apps/telegram_bot/map_links.py
from __future__ import annotations

from urllib.parse import urlencode


def google_maps_url(lat: float, lon: float) -> str:
    return f"https://www.google.com/maps/search/?api=1&query={lat:.6f},{lon:.6f}"


def yandex_maps_url(lat: float, lon: float) -> str:
    point = f"{lon:.6f},{lat:.6f}"
    return "https://yandex.com/maps/?" + urlencode({"ll": point, "z": "16", "pt": f"{point},pm2rdm"})

File: apps/frontend/src/mapLinks.ts
function coordinate(value: number): string {
  if (!Number.isFinite(value)) throw new Error("invalid_coordinate");
  return value.toFixed(6);
}

export function googleMapsUrl(lat: number, lon: number): string {
  return `https://www.google.com/maps/search/?api=1&query=${coordinate(lat)},${coordinate(lon)}`;
}

export function yandexMapsUrl(lat: number, lon: number): string {
  const point = `${coordinate(lon)},${coordinate(lat)}`;
  const query = new URLSearchParams({ ll: point, z: "16", pt: `${point},pm2rdm` });
  return `https://yandex.com/maps/?${query.toString()}`;
}
